src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -127,18 +127,10 @@ object DepositHandler {
     }
   }
 
-  def checkBaseDir()(implicit id: String, baseDir: File): Try[Unit] = {
-    if (!baseDir.exists()) {
-      log.error(s"[$id] Bag store base directory ${baseDir.getAbsolutePath} doesn't exist")
-      Failure(FailedDepositException(id, s"Bag store base directory ${baseDir.getAbsolutePath} doesn't exist"))
-    }
-    else if (!baseDir.canRead) {
-      log.error(s"[$id] Bag store base directory ${baseDir.getAbsolutePath} is not readable")
-      Failure(FailedDepositException(id, s"Bag store base directory ${baseDir.getAbsolutePath} is not readable"))
-    }
-    else {
-      Success(())
-    }
+  def checkBagStoreBaseDir()(implicit id: String, baseDir: File): Try[Unit] = {
```

Comment:
Both logging an error and also throwing it as an exception is often considered and anti-pattern. The reason is that the thrown exception is logged somewhere as well (or at least it should be) and you end up with redundantly logged errors which makes it more difficult to diagnose the problem.